modules/ROOT/pages/db/db-connector-insert-update-delete.adoc

```diff
@@ -3,48 +3,157 @@ ifndef::env-site,env-github[]
+the equivalent query should be written as follows:
+[source,xml,linenums]
+----
+<db:insert config-ref="dbConfig">
+    <db:sql>
+        INSERT INTO electronic(name, description, price, discount)
+        VALUES (:name, :description, :price, :discount)
+    </db:sql>
+    <db:input-parameters>#[{
+        name: 'Coffee Machine',
+        description: 'Model:XYZ99. Uses small size capsules.',
+        price: 120,
+        discount: 5}]
+    </db:input-parameters>
+</db:insert>
+----
+=============
+
+The `insert` operation will output a `Statement Result` object that contains 2 fields: a _affectedRows_
+integer that indicates how many rows were affected by the query, and a _generatedKeys_ map
+which contains the autogenerated keys. Given this example, the output is as follows:
+
+[source,json,linenums]
+----
+{
+  "affectedRows": 1,
+  "generatedKeys": {
+
+  }
+}
+----
+
+The _generatedKeys_ map is empty, although a row is successfully added to the table.
+This is because this information is disabled by default in order to avoid large overheads.
+To activate the information and generate keys, add the `autoGenerateKeys` parameter, and then set it to true:
+
+[source,xml,linenums]
+----
+<db:insert config-ref="dbConfig" autoGenerateKeys="true">
+   <db:sql>
+       INSERT INTO electronic(name, description, price, discount)
+       VALUES (:name, :description, :price, :discount)
+   </db:sql>
+   <db:input-parameters>#[{
+       name: 'Coffee Machine',
+       description: 'Model:XYZ99. Uses small size capsules.',
+       price: 120,
+       discount: 5}]
+   </db:input-parameters>
+</db:insert>
+----
+
+If the statement generates multiple keys per affected row, you can specify which columns
+should be returned by setting the parameter `autoGeneratedKeysColumnNames` to a list that contains
+the desired column names. For example, to ensure only the _id_ column is
+returned, set the parameter to a DataWeave list with a single string element:
+[source,xml,linenums]
+----
+<db:insert config-ref="dbConfig" autoGenerateKeys="true" autoGeneratedKeysColumnNames="#[['id']]">
+    ...
+</db:insert>
+----
+
+== Update Example
+
+Given the table that is defined in the `insert` example, the operation to update the *electronic* table
+to set the discount to 10 percent for all items that have a price above 100 is as follows:
+
+[literal,source,xml,linenums]
+....
 <db:update config-ref="dbConfig">
-  <db:sql>
-    UPDATE PLANET SET DESCRIPTION = :description where POSITION = :position
-  </db:sql>
-  <db:input-parameters>
-  #[
-    {'description' : payload,
-    'position' : 7,
-    }
-  ]
-  </db:input-parameters>
+    <db:sql><![CDATA[#["UPDATE electronic SET discount = :discount WHERE price > :price"]]]></db:sql>
+	<db:input-parameters>#[{
+        discount: 10,
+        price: 100
+    }]</db:input-parameters>
 </db:update>
+....
+
+The `<![CDATA[...]]>` wrapper allows of the use of special characters, such as `>` or `"`, in the query.
```

Comment:
allows of > allows

Note that text vacillates between passive voice and active voice. If you are making a conscious decision to emphasize the action over the actor, then make sure that those cases are clear; otherwise, stick with active voice:

...wrapper enables you to use

Reply:
Went with "allows you to use"... I think enables feels to function-y and technical.